extensions/amp-base-carousel/validator-amp-base-carousel.protoascii

```diff
@@ -0,0 +1,209 @@
+#
+# Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the license.
+#
+
+tags: {  # amp-base-carousel
+  html_format: AMP
+  tag_name: "SCRIPT"
+  extension_spec: {
+    name: "amp-base-carousel"
+    version: "0.1"
+    version: "latest"
+    requires_usage: GRANDFATHERED
+    deprecated_allow_duplicates: true
+  }
+  attr_lists: "common-extension-attrs"
+}
+
+# Examples for media query / value pairs:
+# (min-width: 600px) and (min-height: 800px)  false, true
+# (min-width: 600px) false,(min-width: 600px) true,false
+# true
+attr_lists: {
+  name: "amp-base-carousel-common"
+  attrs: {
+    name: "advance-count"
+    # Media query / integer pairs
+    value_regex: "([^,]+\\s+(-?\\d+),\\s*)*(-?\\d+)"
```

Comment:
The Validator understands the concept of Media Query (see [MediaQuerySpec](https://github.com/ampproject/amphtml/blob/master/validator/validator.proto#L190)) but it's currently only as part of a stylesheet. Would it be worthwhile to extend AttrSpec to have a value that is a MediaQuerySpec and validate against that instead of a regex?

Reply:
The media query itself I'm not too concerned about (and I don't validate here), since an invalid media query just evaluates to false when I go to run it, I want to mainly validate that it is a comma separated list, with one single value at the end. I think it would be useful to be able to have a format field which you could do something like:

```
value_format: "media_query_value_pairs"
value_format_item_regex: "(-?\\d+)"
```

Reply:
I have been thinking about a DeliminatorSpec that would define a deliminator and then the set of values to apply to each item. Seems like there may be something there. Since it's not necessary for this PR will continue to just think about it and ping you if/when it makes progress.